### PR TITLE
Make classes with only a single funtion free standing functions

### PR DESCRIPTION
**Pull request by daniel-j-h**
- [X] DouglasPeuker
- [x] PolylineCompressor
- [x] PolylineFormatter
- [x] RouteNameExtraction
- [x] TurnInstructionsClass
- [x] ComputeAngle

### CODE REVIEW

File: src/engine/douglas_peucker.cpp [hunk 1]
```diff
@@ -59,13 +59,15 @@ struct CoordinatePairCalculator
 };
 }
 
-void DouglasPeucker::Run(std::vector<SegmentInformation> &input_geometry, const unsigned zoom_level)
+void DouglasPeucker(std::vector<SegmentInformation>::iterator begin,
```
Comment (TheMarex) [inline src/engine/douglas_peucker.cpp]: Leading char should be small case like: `douglasPeucker` for free functions.

Comment (daniel-j-h) [inline src/engine/douglas_peucker.cpp]: Urgh yeah have to get used to that again. Also having different conventions between member functions and free standing function is somewhat ugly for refactoring :/

**Comment by daniel-j-h** [comment]
Done. Converted to free standing functions. At the same time:

- fixing includes in the files I touched
- fixing two occurrences of Undefined Behavior when instantiating templates with incomplete type
- made sure compile-time constants are guaranteed to be ROM-able

/cc @TheMarex

**Comment by TheMarex** [comment]
@daniel-j-h seems like it broke builds on windows. Investigating.

**Comment by daniel-j-h** [comment]
> 'not': undeclared identifier

https://github.com/Project-OSRM/osrm-backend/pull/1871

**Comment by TheMarex** [comment]
Rebased on #1871, merge after that PR.

**Pull request merged**